### PR TITLE
patch_manager: warning for incompatible patches

### PR DESCRIPTION
**Pull request by Megamouse**
This will open a warning dialog whenever the patch manager is opened and incompatible patches are detected.

image

### CODE REVIEW

**Comment by Megamouse** [comment]
Removed the note about legacy patches, since legacy patches aren't supposed to be located in the patches folder anyway.

File: rpcs3/rpcs3qt/patch_manager_dialog.cpp [hunk 1]
```diff
@@ -135,9 +136,25 @@ void patch_manager_dialog::load_patches()
...
+	}
+
+	if (show_error && has_errors)
+	{
+		// Open a warning dialog after the patch manager was opened
+		QTimer::singleShot(100, [this]()
+		{
+			QMessageBox::warning(this, tr("Incompatible patches detected"),
+				tr("Some of your patches are not compatible with the current version of RPCS3's Patch Manager.\n\nMake sure that all the patches located in \"%0\" contain the proper formatting that is required for the Patch Manager Version %1.")
+				.arg(QString::fromStdString(patch_engine::get_patches_path())).arg(QString::fromStdString(patch_engine_version)));
```
Comment (sampletext32) [inline rpcs3/rpcs3qt/patch_manager_dialog.cpp:156]: You have a const a bit before, so don't need to call get_patches_path()
```suggestion
				.arg(QString::fromStdString(patches_path)).arg(QString::fromStdString(patch_engine_version)));
```

Comment (Megamouse) [reply rpcs3/rpcs3qt/patch_manager_dialog.cpp:156]: Ah thx. I had some extra variable set to replace my name in the path and forgot to reuse the proper one in the final version.

Comment (Megamouse) [reply rpcs3/rpcs3qt/patch_manager_dialog.cpp:156]: Can you also make a suggestion for the capture? I'm on the train and can't properly work on this on my own right now

Comment (sampletext32) [inline rpcs3/rpcs3qt/patch_manager_dialog.cpp:156]:
```suggestion
				.arg(QString::fromStdString(patches_path)).arg(QString::fromStdString(patch_engine_version)));
```

Comment (Megamouse) [reply rpcs3/rpcs3qt/patch_manager_dialog.cpp:156]: This is the same as before.
I just need the proper capture to make it compile. I can batch merge single line suggestions if that's what you worry about

File: rpcs3/rpcs3qt/patch_manager_dialog.cpp [hunk 2]
```diff
@@ -135,9 +136,25 @@ void patch_manager_dialog::load_patches()
...
+		if (!patch_engine::load(m_map, patches_path + path.toStdString()))
+		{
+			has_errors = true;
+		}
+	}
+
+	if (show_error && has_errors)
+	{
+		// Open a warning dialog after the patch manager was opened
+		QTimer::singleShot(100, [this]()
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/patch_manager_dialog.cpp:152]:
```suggestion
		QTimer::singleShot(100, [this, patches_path]()
```
I realized I can do it on my own xD

**Comment by jy95** [comment]
@Megamouse Useful to log the files with errors in the console ?

**Comment by Megamouse** [comment]
The logs already contain that information

**Comment by GarettJaxx** [comment]
Is there any way to make the option "Only Show Owned Games" permanent when you select it?

Always seems to reset to 'off' when I click it on then shut down. No big deal - just asking....

**Pull request merged**